syntax = "proto3";

service Controller {
    rpc StartServer(StartArgs) returns (StartReply);
    rpc StopServer(StopArgs) returns (StopReply);
    rpc GetAll(ServersArgs) returns (ServersReply);
    rpc KillAll(ServersArgs) returns (StopReply);
}

// Request message for getting the running servers
message StartArgs {
    int32 Index = 1;
    repeated string Ips = 2;
}

// Response message for starting the server
message StartReply {
}

// Request message for stopping the server
message StopArgs {
    string Ip = 1;
}

// Response message for stopping the server
message StopReply {
}

// Response message for getting the running servers
message ServersArgs {
}

// Response message for getting the running servers
message ServersReply {
    repeated string Ips = 1;
}
